File: src/main/java/kz/otussocialnetwork/service/AuthService.java
package kz.otussocialnetwork.service;

import kz.otussocialnetwork.model.dto.AuthResponse;
import kz.otussocialnetwork.security.model.Authentication;
import lombok.NonNull;

public interface AuthService {

  AuthResponse login(@NonNull String username,
                     @NonNull String password);

  AuthResponse refreshToken(@NonNull String accessToken);

  Authentication getAuthentication(@NonNull String accessToken);
}

File: src/main/java/kz/otussocialnetwork/service/AuthService.kt
package kz.otussocialnetwork.service

import kz.otussocialnetwork.model.dto.AuthResponse
import kz.otussocialnetwork.security.model.Authentication

interface AuthService {
    fun login(
        username: String,
        password: String
    ): AuthResponse

    fun refreshToken(refreshToken: String): AuthResponse

    fun getAuthentication(accessToken: String): Authentication
}
